lib/uiautomator2.js

```diff
@@ -26,36 +28,80 @@ class UiAutomator2Server {
     });
   }
 
-  async installServerApk () {
+  async installServerApk (installRetries) {
     // Installs the apks on to the device or emulator
     let apkPackage = await this.getPackageName(apkPath);
     // appending .test to apkPackage name to get test apk package name
-    let testApkPackage = apkPackage + '.test';
+    let testApkPackage = `${apkPackage}.test`;
     let isApkInstalled = await this.adb.isAppInstalled(apkPackage);
     let isTestApkInstalled = await this.adb.isAppInstalled(testApkPackage);
     if (isApkInstalled || isTestApkInstalled) {
       //check server apk versionName
       let apkVersion = await this.getAPKVersion(apkPath);
       let pkgVersion = await this.getInstalledPackageVersion(apkPackage);
       if (apkVersion !== pkgVersion) {
-        isApkInstalled = false;
-        isTestApkInstalled = false;
-        await this.adb.uninstallApk(apkPackage);
-        await this.adb.uninstallApk(testApkPackage);
+        logger.debug(`Server installed but version ${pkgVersion} instead of ${apkVersion}. Re-installing`);
+        isApkInstalled = isTestApkInstalled = false;
+        await this.uninstallServerAndTest(apkPackage, testApkPackage);
       }
     }
+
     if (!isApkInstalled) {
       await this.signAndInstall(apkPath, apkPackage);
     }
     if (!isTestApkInstalled) {
-      await this.signAndInstall(testApkPath, testApkPackage);
+      await this.signAndInstall(testApkPath, testApkPackage, true);
+    }
+
+    let retries = getRetries('Server install', installRetries, SERVER_INSTALL_RETRIES);
+
+    logger.debug(`Waiting up to ${retries * 1000}ms for instrumentation '${INSTRUMENTATION_TARGET}' to be available`);
+    let output;
+    try {
+      await retryInterval(retries, 1000, async () => {
+        try {
+          output = await this.adb.shell(['pm', 'list', 'instrumentation']);
+          if (output.indexOf('Could not access the Package Manager') !== -1) {
+            output = null; // remove output, so it is not printed below
+            throw new Error(`Problem running package manager: ${output}`);
+          } else if (output.indexOf(INSTRUMENTATION_TARGET) === -1) {
+            throw new Error('No instrumentation process found. Retrying...');
+          }
+          logger.debug(`Instrumentation '${INSTRUMENTATION_TARGET}' available`);
+        } catch (err) {
```

Comment:
the fact that we do nothing with `err` other than throw it makes me think we don't need this to be in a try/catch block.